fawltydeps/extract_declared_dependencies.py

```diff
@@ -382,42 +367,29 @@ def validate_deps_source(
     - Raise UnparseablePathException if the given path cannot be parsed.
 
     The given 'parser_choice' and 'filter_by_parser' change which files we
-    consider valid sources:
-
-    - If parser_choice is None, we will consider any file path that matches one
-      of the PARSER_CHOICES above a valid file.
-    - If parser_choice is not None, and filter_by_parser is False, we assume
-      that the given file path was explicitly passed by the user, and meant to
-      be parsed with the given parser_choice, _even_ when the file path does not
-      nominally match this parser choice: We return the file path as a valid
-      source regardless of its file name/suffix.
-    - If parser_choice is not None, and filter_by_parser is True, we assume that
-      the given file path was encountered while walking a directory passed by
-      the user. In this case, we will only consider the file valid if it matches
-      the given parser choice.
+    consider valid sources: With parser_choice=None, any file path that matches
+    one of the PARSER_CHOICES above is considered a valid file. Otherwise,
+    filter_by_parser is used to either consider only files that match the given
+    parser_choice (filter_by_parser=True), or to force all files to be parsed as
```

Comment:
I meant all files passed to this function with `filter_by_parser=False`. When looking at this function together with its caller (`find_sources()`) this ends up being all files passed directly via `.deps`. Any files encountered indirectly (while walking a directory) are passed with `filter_by_parser=True` and are thus not covered by this.

Since this docstring is already inside the context of `validate_deps_source()` which only takes a single path at a time, this phrasing is not as clear as it should be. Will rewrite to:
```
    The given 'parser_choice' and 'filter_by_parser' determine which file paths
    we consider valid sources, and how they are parsed: With parser_choice=None,
    a file path will use the first matching parser in PARSER_CHOICES above, if
    any. Otherwise - when parser_choice is specified - the file must either
    match this parser (filter_by_parser=True), or this parser will be forced
    even if the file does not match (filter_by_parser=False).
```

Reply:
Fixed in 250d980.